### PR TITLE
Add intentions doc

### LINKED ISSUE #42: Add intentions documentation
## What is the issue?
No documentation about intentions.

## Suggestion(s)/solution(s) [Optional]
Add intetions documentation in README.md

## Checklist (after created issue)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### PR DESCRIPTION
**Pull request by zhenik**
## Closes/fixes/resolves issue(s)?
- closes #42 
## What was added/changed/fixed?
- [updated] doc

## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [x] Added to project
- [x] Added to milestone

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,32 +6,65 @@
...
+make up
+```
 
 ### Providers
 - [Nomad](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs)
 - [Vault](https://registry.terraform.io/providers/hashicorp/vault/latest/docs)
 
+### Intentions
+Service mesh [communication controlled by intentions](https://learn.hashicorp.com/tutorials/consul/get-started-service-networking#control-communication-with-intentions).
+Intentions are required when [consul acl is enabled and consul_acl_default_policy is deny](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).
```
Comment (pdmthorsrud) [inline README.md:55]:
```suggestion
Intentions are required when [`consul_acl=true` and `consul_acl_default_policy=deny`](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).
```

Comment (zhenik) [reply README.md:55]: This module can be used without vagrant-hashistack, so I would like to decouple from its context in the documentation. WDYT?

```suggestion
Intentions are required when [consul acl is enabled and default_policy is deny](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents).
```

Comment (pdmthorsrud) [reply README.md:55]: Oh right, that is a very good point. Didn't think of that at all.

**Review by pdmthorsrud — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -170,23 +203,12 @@ OK
...
-FIELDS TERMINATED BY ','
-LINES TERMINATED BY '\n'
-LOCATION 's3a://hive/some/prefix/'
-TBLPROPERTIES ("skip.header.line.count"="1");
-```
-Query table `iris`
-```sql
-SELECT * FROM default.iris LIMIT 10;
-```
+
```
Comment (claesgill) [inline README.md:206]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -6,32 +6,65 @@
...
+```sh
+make up
+```
 
 ### Providers
 - [Nomad](https://registry.terraform.io/providers/hashicorp/nomad/latest/docs)
 - [Vault](https://registry.terraform.io/providers/hashicorp/vault/latest/docs)
 
+### Intentions
+Service mesh [communication controlled by intentions](https://learn.hashicorp.com/tutorials/consul/get-started-service-networking#control-communication-with-intentions).
```
Comment (claesgill) [inline README.md:54]: Can you make this a more descriptive sentence? I'm not sure what this means.

File: README.md (hunk 2 above)
Comment (zhenik) [reply README.md:206]: I am not sure that I understand ur suggestion.

File: README.md (hunk 3 above)
Comment (zhenik) [reply README.md:54]: This is straight forward IMHO.
```
Service mesh communication controlled by intentions. 
Intentions are required when consul acl is enabled and default_policy is deny. 
```

Did u mean part about `Service mesh` (WHAT is it? WHY? LINKS?) or smth else?

File: README.md (hunk 2 above)
Comment (pdmthorsrud) [reply README.md:206]: I think he just means to remove one line for formatting's sake

Comment (zhenik) [reply README.md:206]: ahhh :)

File: README.md (hunk 3 above)
Comment (claesgill) [reply README.md:54]: More like a build-up to why I would use that link.

Comment (zhenik) [reply README.md]: Is it better described in https://github.com/fredrikhgrelland/terraform-nomad-hive/pull/43/commits/cfe214db12c728da9c672076ed366b3cccbe9a2f ?

Comment (pdmthorsrud) [reply README.md]: Just to chime in, I like the one you linked better, Nikita. It gives some small context.

**Review by pdmthorsrud — APPROVED** [review]
LGTM

**Review by claesgill — APPROVED** [review]
LGTM!

**Pull request merged**